addons/gaea/graph/graph_nodes/root/vector/operations/vector_operation.gd

```diff
@@ -0,0 +1,143 @@
+@tool
+class_name GaeaNodeVectorOp
+extends GaeaNodeVectorBase
+## Base class for operations between 2 vectors.
+
+enum Operation {
+	Add,
+	Subtract,
+	Multiply,
+	Divide,
+}
+
+
+class Definition:
+	var args: Array[StringName]
+	var output: String
+	var conversion: Callable
+	func _init(_args: Array[StringName], _output: String, _conversion: Callable):
+		args = _args
+		output = _output
+		conversion = _conversion
+
+
+var OPERATION_DEFINITIONS: Dictionary[Operation, Definition] : get = _get_operation_definitions
+
+
+func _get_title() -> String:
+	return "VectorOp"
+
+
+func _get_description() -> String:
+	match get_enum_selection(0):
+		Operation.Add:
+			return "Sums 2 [code]Vector[/bg][/c]."
+		Operation.Subtract:
+			return "Subtracts 2 [code]Vector[/bg][/c]."
+		Operation.Multiply:
+			return "Multiplies 2 [code]Vector[/bg][/c] together."
+		Operation.Divide:
+			return "Divides 2 [code]Vector[/bg][/c] together."
+	return ""
+
+
+func _get_tree_items() -> Array[GaeaNodeResource]:
+	var items: Array[GaeaNodeResource]
+	for vector_type in VectorType.keys():
+		for operation in OPERATION_DEFINITIONS.keys():
+			var item: GaeaNodeResource = get_script().new()
+			item.set_tree_name_override("%s %s (%s)" % [
```

Comment:
Tree names should be PascalCase so there shouldn't be a space between the first 2 `%s`s. (The parentheses are an exception since those add information).

Reply:
Updated